### DISCUSSION TITLE
[ERROR] Error Downloading/Verifying Codebase

### DISCUSSION THREAD

**Discussion by Vpret** [nextgenhealthcare/connect#5433] [Q&A]
Hello Hubs,

We are facing some issue with mirth, can anyone help in this, need urgently,

While loading Main JNLP

Unable to retrieve payload from HTTP request. URL:http://localhost:___/webstart
java.lang.Exception:unable to retrieve payload from HTTP request. URL:http://localhost:___/webstart
at com.mirth.connect.client.launcher.f.a.(sourcefile:791)
at com.mirth.connect.client.launcher.f.a.(sourcefile:671)
at com.mirth.connect.client.launcher.f.a.(sourcefile:167)
etc....

**Comment by pacmano1** [comment]
Is the enging running?
Does netstat show the port is listening?
Does mirth.log show anything?

You have posted no details.  Much like saying "my car won't start".

**Reply by pacmano1** [reply]
Moved to discussions until @Vpret  provides more details.

**Reply by pras02** [reply]
Hi, I have a similar issue.
engine is running and the ports are listening and well.

Below is the specific error that I see, I have never set any user name password yet anywhere during the installation.

Caused by: java.lang.Exception: HTTP request returned invalid response. HTTP status: HTTP/1.1 403 Forbidden, URI: http://localhost:8080/webstart
	at com.mirth.connect.client.launcher.f.a(SourceFile:777)
	... 4 more

**Reply by pras02** [reply]
[Codebase Downloader Thread] com.mirth.connect.client.launcher.MirthClientLauncher: java.lang.Exception: Unable to retrieve payload from HTTP request. URI: http://localhost:8080/webstart

**Reply by pacmano1** [reply]
If you are using the Mirth Admin Launcher (you should be), use `http://<ip of server>:8080/`.  But you should really be using `https://<ip of server>:8443/`

**Reply by pras02** [reply]
Tried using the local IP address with all possible combinations of http/https but no luck. Service is running locally on my laptop so even 127.0.0.1 did not help. and yes I am using the Admin Launcher.

**Reply by pacmano1** [reply]
what does netstat show?

**Reply by pras02** [reply]
TCP    0.0.0.0:8443           0.0.0.0:0              LISTENING
  TCP    [::]:8443              [::]:0                 LISTENING
  
  TCP    0.0.0.0:8080           0.0.0.0:0              LISTENING
  TCP    [::]:8080              [::]:0                 LISTENING

**Reply by pacmano1** [reply]
and if you open a web browser to `https://<use your real ip not localhost>:8443/webadmin/Index.action`

What do you see?

**Reply by pras02** [reply]
![image](https://user-images.githubusercontent.com/52795849/216707747-e978a82c-e189-4d07-98fa-c25579a70ded.png)
 I do see this page and clicking on the Launch Mirth Connect Administrator downloads the .jnlp file
 
 Clicking that gives me the same error. This time I am using my local IP address of the system.
 Unable to retrieve payload from HTTP request. URI: https://10.0.0.153:8443/webstart
java.lang.Exception: Unable to retrieve payload from HTTP request. URI: https://10.0.0.153:8443/webstart
	at com.mirth.connect.client.launcher.f.a(SourceFile:788)
	at com.mirth.connect.client.launcher.f.a(SourceFile:668)
	at com.mirth.connect.client.launcher.f.a(SourceFile:168)
	at com.mirth.connect.client.launcher.MirthClientLauncher.run(SourceFile:1286)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.Exception: HTTP request returned invalid response. HTTP status: HTTP/1.1 401 Unauthorized, URI: https://10.0.0.153:8443/webstart
	at com.mirth.connect.client.launcher.f.a(SourceFile:777)
	... 4 more
	
![image](https://user-images.githubusercontent.com/52795849/216708224-06dc2a72-d91d-4758-ae50-2908ce363b94.png)

**Reply by pacmano1** [reply]
So your are using the Mirth Admin Launcher?

![image](https://user-images.githubusercontent.com/44065187/216708247-f260f41a-c396-445f-8a9a-64a6050139ee.png)

**Reply by pras02** [reply]
I am wondering why it says 401 Unauthorized even before login prompt. I can reinstall and try but just wondering. Also, I am running Java 8 which is supported as per the guide. Can't understand the reason behind this error.

**Reply by pras02** [reply]
> So your are using the Mirth Admin Launcher?
> 
yes I am doing the exact same thing from the launcher with my local IP address.

**Reply by pacmano1** [reply]
You sure some other process is not running?   Stop the mirth server process, confirm it is stopped.  Does netstat show ports not listening?

Also try a wget like below.  Ignore mine saving as index.html.2, there was another local file.

![image](https://user-images.githubusercontent.com/44065187/216711456-8745a53f-a78d-4c03-b57d-c860b18c51c0.png)

**Reply by pras02** [reply]
yes, stopped the service and checked that no other process is running on 8443 and 8080. 
FYI, I just reinstalled as well but the still the same.

wget cmd output
![image](https://user-images.githubusercontent.com/52795849/216713107-41870d61-dda5-47e3-92c9-8078b76d5a2e.png)

**Reply by pacmano1** [reply]
Please try the admin launcher with your actual LAN ipv4, not localhost (albeit that really should work).

**Reply by pras02** [reply]
Same setup file I was just able to install on another laptop and got it working without any issue in minutes (able to login, also used localhost and it works perfectly fine). It is very strange to see this error on this particular machine and unable to figure out the reason. I will keep trying and let you know if I succeed in getting it working here. Thanks for all the help and suggestions buddy!

**Comment by pras02** [comment]
Finally got it working by changing the default ports (8080 and 8443) in the C:\Program Files\Mirth Connect\conf\mirth.Properties file to 9090 and 9443. 

http.port = 9090
https.port = 9443

I am 100% sure that there is no other application/service running on 8080 and 8443, but somehow the Mirth service did not like it but always showed that the server started with those ports. Ideally, I would expect the service to not run and log stating the ports are in use by another program. So, in this case the server logs were misleading saying it started it on 8080 or 8443.

**Reply by pacmano1** [reply]
That's bizarre.   Glad you got it working though.
